src/Repositories/AbstractModelRepository.php

```diff
@@ -6,73 +6,79 @@
 
 use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Support\LazyCollection;
 use Wimski\ModelRepositories\Contracts\Repositories\ModelRepositoryInterface;
 
 /**
- * @template T of \Illuminate\Database\Eloquent\Model
- * @implements ModelRepositoryInterface<T>
+ * @template TModel of Model
+ * @implements ModelRepositoryInterface<TModel>
  */
 abstract class AbstractModelRepository implements ModelRepositoryInterface
 {
     /**
-     * @var T
+     * @var TModel
      */
-    protected $model;
+    protected Model $model;
 
-    public function builder(): Builder
+    public function builder(bool $withGlobalScopes = true)
     {
-        /** @var Builder<T> $builder */
-        $builder = $this->model->newQuery();
+        /** @var Builder<TModel> $builder */
```

Comment:
Because of this :sob: 
```
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/Repositories/AbstractModelRepository.php                                                                                                                                             
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  31     Parameter #1 $builder of method Illuminate\Database\Eloquent\Model::registerGlobalScopes() expects Illuminate\Database\Eloquent\Builder, Illuminate\Database\Eloquent\Builder<TModel of  
         Illuminate\Database\Eloquent\Model>|TModel of Illuminate\Database\Eloquent\Model given.                                                                                                  
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

```